Fix: Append extensions only when saving

gDanix opened the pull request:
The only thing I do in this PR is to enclose all the extension-processing code in a block that executes only when the dialog is opened for saving.

This fixes the bug introduced and described in #6401 

Because of the deeper indent, GitHub shows many meaningless additions and deletions.

rwjuk commented:
When fixing a logged issue, can you please make your commit message something like `Fix #6401: Append extensions only when saving`. I've not tested the actual code yet, I'll do that in a bit.

gDanix commented:
Actually, this PR doesn't fix any issue, that's why I didn't specified the issue number.
#3510 was fixed in #6401.
This PR is fixing a bug I introduced and described (after merge) in #6401, so I fixed it as soon as I discovered it, instead of creating and issue

rwjuk commented:
@gDanix That'll teach me to read the history fully!! You're quite right.

gDanix commented:
Nevermind :)
